src/index.ts

```diff
@@ -71,40 +40,109 @@ function idToAsset(id: string): AssetID {
   }
 }
 
-async function assetToAccount(asset: AssetID, timestamp: string, rpcUrl: string): Promise<AccountID> {
-  const owner = await erc721OwnerOf(asset, rpcUrl)
-  return new AccountID({
-    chainId: asset.chainId,
-    address: owner
-  })
+async function assetToAccount(asset: AssetID, timestamp: number, customSubgraph?: SubGraphUrls): Promise<AccountID[]> {
+  // we want to query what block is at the timestamp IFF it is an (older) existing timestamp
+  let queryBlock: number = undefined;
+  if (timestamp && !isWithinLastBlock(timestamp)) {
+    queryBlock = await blockAtTime(timestamp);
+  }
+
+  let owners: string[];
+  if (asset.namespace === ErcNamespace.ERC721) {
+    owners = [ await erc721OwnerOf(asset, queryBlock, customSubgraph?.erc721) ];
+  } else {
+    owners = await erc1155OwnersOf(asset, queryBlock, customSubgraph?.erc1155);
+  }
+
+  return owners.slice().map(owner => 
+    new AccountID({
+      chainId: asset.chainId,
+      address: owner
+    })
+  );
 }
 
-async function accountToDid(account: AccountID, atTime: number, ceramic: CeramicApi): Promise<string | null> {
+async function createCaip10Link(account: AccountID, ceramic: CeramicApi): Promise<string | null> {
   const doc = await ceramic.createDocument('caip10-link', {
     metadata: {
       family: 'caip10-link',
       controllers: [AccountID.format(account)]
     }
-  })
+  });
   // TODO - enable a way to do this with one request
   //const docAtTime = await ceramic.loadDocument(doc.id, { atTime })
-  return doc?.content
+  return doc?.content;
```

Comment:
```suggestion
// pseudo code
  if(timestamp) {
    const docAtTime = await ceramic.loadDocument(doc.id, { atTime: timestamp })
    return docAtTime?.content
  } else {
    return doc?.content
  }
```
We also need to use the timestamp here to load the caip10-link at the correct point in time.

Reply:
Right that's quite an important part! Thanks for catching that.